### PR TITLE
replace GPG signing recommendation with SSH

### PR DESCRIPTION
**Pull request by audreyality**
## 🎟️ Tracking

N/A

## 📔 Objective

Now that GitHub's SSH key signing support has matured, recommend using an SSH key for commit verification.

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: docs/contributing/commit-signing.mdx [hunk 1]
```diff
@@ -21,34 +21,38 @@ their commits.
...
-1.  Install GnuPG:
+:::tip
 
-  <Tabs groupId="os">
+If you're unsure what to use, we recommend you create a commit signing key using SSH. Remember to
+protect the key with a [strong passphrase or password][password-generation].
 
-    <TabItem value="mac" label="macOS">
+```bash
+ssh-keygen -f ~/.ssh/bw-signing -C "$(git config --get user.email)" -t ed25519
```
Comment (withinfocus) [inline docs/contributing/commit-signing.mdx:32]: ❓ Are we implying anything with the unique name here? Versus just using the default and this be one signing key to rule them all.

File: docs/contributing/commit-signing.mdx [hunk 2]
```diff
@@ -21,34 +21,38 @@ their commits.
...
+1. Configure git to sign using SSH.
 
-  </Tabs>
+   ```bash
+   git config --global gpg.format=ssh
+   git config --global user.signingkey=~/.ssh/bw-signing.pub
+   ```
 
-2.  Follow the [Github documentation][github-verification] to configure commit signing
+   (Remove the `--global` flag to only apply this setting to the current repository.)
```
Comment (withinfocus) [inline docs/contributing/commit-signing.mdx:46]: 💭 Seems like overkill to _not_ have this global so I wouldn't even mention it.

**Review by withinfocus — APPROVED** [review]
Works for me, and something I am already doing personally.

File: docs/contributing/commit-signing.mdx (hunk 1 above)
Comment (audreyality) [reply docs/contributing/commit-signing.mdx:32]: Your auth key to sign in to many computers in an interactive online orocess. Your signing key, meanwhile, only signs github commits, which is an offline process. Compartmentalizing the use-cases is best practice.

This is also a prelude to recommending keys loaded from Bitwarden, once we've released SSH support. In such a case, we'd need to identify the vault item containing the key.

File: docs/contributing/commit-signing.mdx (hunk 2 above)
Comment (audreyality) [reply docs/contributing/commit-signing.mdx]: Addressed by 616af63a5.

File: docs/contributing/commit-signing.mdx (hunk 1 above)
Comment (audreyality) [reply docs/contributing/commit-signing.mdx]: [72d55e5](https://github.com/bitwarden/contributing-docs/pull/507/commits/72d55e51863d071820cc5c6007b9077600087a11) extracts the path into a variable that's shared with the git configuration sample. This should better communicate that the the filename is arbitrary. 😄

**Review by withinfocus — APPROVED** [review]

**Pull request merged**